dxp/javascript_minification.md

```diff
@@ -0,0 +1,86 @@
+# JavaScript Minification in DXP
+
+This article describes DXP infrastructure to minify JavaScript source files. It is related to [Resource Serving in DXP](resource_serving.md).
+
+## MinifierUtil, JavaScriptMinifier and frontend-js-minifier
+
+These three pieces conform the basic infrastructure available in DXP to minify JavaScript code.
+
+-   [JavaScriptMinifier](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/minifier/JavaScriptMinifier.java): describes the interface that services to minify JavaScript must implement.
+
+-   [MinifierUtil](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/minifier/MinifierUtil.java): provides a way to retrieve the `JavaScriptMinifier` with highest service ranking currently deployed to DXP. Note that you can also obtain a reference to a `JavaScriptMinifier` (or all of them) using standard OSGi mechanisms.
+
+-   [frontend-js-minifier](https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-js/frontend-js-minifier): this project contains the two default minifiers that DXP ships out of the box (Google and Yahoo). See the two following sections for more details.
```

Comment:
"out of the box" → "out-of-the-box"

Reply:
How do the `portal-impl` classes access these `module` minifiers? I might be wrong here, but I assume we didn't remove the minifiers in `portal-impl`... could we make some notes about that, just to clarify what's being used and what isn't?

Reply:
> How do the portal-impl classes access these module minifiers

Through `MinifierUtil`, because it is in `portal-impl`.

Reply:
I mean... we have a static `util` in `portal-impl` accessing OSGi Services?

Reply:
I see we don't have them in `portal-impl` anymore, though, so there's no duplication which was my main source of confusion.

Reply:
I added some text explaining it.

Reply:
> I mean... we have a static util in portal-impl accessing OSGi Services?

Yes, that was the usual way of OSGifying things in the past without breaking the API. That's the reason why OSGi `Registry` was created: to bridge outside-of-OSGi code and OSGi world.